<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:unless value="$(arg sim_mode)">
        <ros2_control name="RealRobot" type="system">
            <hardware>
                <plugin>odrive_ros2_control_plugin/ODriveHardwareInterface</plugin>
                <param name="can">can1</param>
            </hardware>
            <joint name="arm_joint_1">
                <param name="node_id">11</param>
                <param name="transmission">26</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            
            <joint name="arm_joint_2">
                <param name="node_id">12</param>
                <param name="transmission">26</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_3">
                <param name="node_id">13</param>
                <param name="transmission">26</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_4">
                <param name="node_id">14</param>
                <param name="transmission">26</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_5">
                <param name="node_id">15</param>
                <param name="transmission">26</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

       

        </ros2_control>
    </xacro:unless>

    <xacro:if value="$(arg sim_mode)">
        <ros2_control name="GazeboSystem" type="system">
            <hardware>
                <plugin>gz_ros2_control/GazeboSimSystem</plugin>

            </hardware>

            <joint name="arm_joint_1">
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            
            <joint name="arm_joint_2">
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_3">
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_4">
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="arm_joint_5">
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>


        </ros2_control>

    </xacro:if>

    <gazebo>
        <plugin name="gz_ros2_control::GazeboSimROS2ControlPlugin" filename="libgz_ros2_control-system.so">
            <parameters>$(find manip)/config/controllers.yaml</parameters>
            <parameters>$(find manip)/config/gaz_ros2_ctl_use_sim.yaml</parameters>
        </plugin>
    </gazebo>
   
 

 


</robot>
